java/README.md

```diff
@@ -0,0 +1,15 @@
+# Java examples
```

Comment:
This feels like repitition of the root README?

Reply:
W/o it there is no overview when you browse the java folder (e.g. on Github). I think this is not ideal. Moreover, our current root README is not really hard to parse and read. I think this is something we need to improve.